website/app/components/doc/link-with-icon/index.hbs

```diff
@@ -4,13 +4,26 @@
 }}
 
 {{#if @route}}
-  <LinkTo class={{this.classNames}} @route={{@route}} @models={{this.models}} ...attributes>
+  <LinkTo
+    class={{this.classNames}}
+    @route={{@route}}
+    @models={{this.models}}
+    ...attributes
+    {{on "click" (fn this.eventTracking.trackEvent @eventName)}}
```

Comment:
@Dhaulagiri while discussing modifiers with @aklkv he mentioned that this could be a [custom modifier](https://github.com/ember-modifier/ember-modifier) (it would make the code cleaner). He can probably expand on this more, but something to consider.

Reply:
ooh, that sounds interesting. Admittedly I built this based on my very stale ember knowledge, so if there is a more idiomatic, current way to do this I'm open to refactoring this.

Reply:
[Opened a task to investigate this suggestion](https://hashicorp.atlassian.net/browse/HDS-3351)